ast/parser.go

```diff
@@ -343,7 +343,20 @@ func (p *Parser) Parse() ([]Statement, []*Comment, Errors) {
 		break
 	}
 
-	if p.po.ProcessAnnotation {
+	selfImported := false
+	// Only look for self imports if necessary
+	if !p.po.ProcessAnnotation {
+		for i := 0; i < len(stmts) && !selfImported; i++ {
+			switch stmt := stmts[i].(type) {
+			case *Import:
+				if isSelfImport(stmt) {
+					selfImported = true
+				}
+			}
+		}
+	}
```

Comment:
A drawback of this approach is that modules not importing `future.self` won't get annotations parsed even if they're part of the package ancestry of a module that has that import.

Another drawback is that we need to find some other solution once `self` is promoted out of the `future` category.

Reply:
💭 I wonder if we could change ProcessAnnotation to be _true_ by default. So, you could still opt-out, if need be, but in general, the annotations would just be available. It might simplify things a lot?

I suppose there's at least one back-compat problem here: invalid annotations would not cause an issue with the current release, but with the next one they would break things, if annotations are parsed by default. Maybe one way would would be to parse them by default, but don't raise the error when ProcessAnnotation is false...? 🤔

Reply:
Yea, that's not a bad idea. We'd just not produce that specific set of annotations for you if they're invalid.

I think another concern with having it always enabled is performance.